Automated cherry pick of #80465 #83250: remove apiserver loopback client QPS limit

liggitt opened the pull request:
Cherry pick of #80465 on release-1.14.

#80465: remove apiserver loopback client QPS limit
#83250: deflake TestWatchBasedManager

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

answer1991 committed:
remove apiserver loopback client QPS limit

lavalamp commented:
/lgtm

lavalamp commented:
Justification: Running out of self-request tokens (when checking legacy token authn, and possibly at other times) puts apiserver into a very mysterious priority inversion.

lavalamp commented:
(In theory 1.14 has made the token auth path not make these requests, but to be consistent with the pick to 1.13, pick it here too. Plus there could be other paths that silently hit this.)

liggitt commented:
/retest
/kind bug
/sig api-machinery
/priority important-soon

tpepper commented:
/retest

tpepper commented:
/retest

liggitt commented:
/hold

Looks like that test is consistently timing out. Taking a look.

liggitt committed:
Set QPS and Burst in integration test

liggitt commented:
that integration test was taking the server loopback config as a starting point, and increasing QPS. now that the loopback config does not set QPS or Burst, the integration test needs to override both. Pushed a commit fixing the test.

liggitt commented:
fixed in master in https://github.com/kubernetes/kubernetes/pull/83250, updated description to include both PRs this is picking

/hold cancel
/retest

lavalamp commented:
/lgtm
